app/backend/src/couchers/interceptors.py

```diff
@@ -34,26 +70,43 @@ class AuthValidatorInterceptor(grpc.ServerInterceptor):
     terminates the call with an UNAUTHENTICATED error code.
     """
 
-    def __init__(self, get_session_for_token, allow_jailed=True):
-        self._get_session_for_token = get_session_for_token
-        self._allow_jailed = allow_jailed
+    def __init__(self):
+        self._pool = get_descriptor_pool()
 
     def intercept_service(self, continuation, handler_call_details):
-        token = parse_session_cookie(dict(handler_call_details.invocation_metadata))
+        method = handler_call_details.method
+        # method is of the form "/org.couchers.api.core.API/GetUser"
+        _, service_name, method_name = method.split("/")
 
-        if not token:
-            return unauthenticated_handler()
+        service_options = self._pool.FindServiceByName(service_name).GetOptions()
+        auth_level = service_options.Extensions[annotations_pb2.auth_level]
 
-        # None or (user_id, jailed)
-        res = self._get_session_for_token(token=token)
+        # if unknown auth level, then it wasn't set and something's wrong
+        if auth_level == annotations_pb2.AUTH_LEVEL_UNKNOWN:
+            raise Exception("Service not annotated with auth_level")
```

Comment:
We should make a unit test that checks that all functions are annotated and remove AUTH_LEVEL_UNKNOWN and this run-time check.

Reply:
Protobuf says that each enum should have an `_UNKNOWN` or equivalent.

Happy to add unit tests, but I'd be more comfortable with this in place just as a sanity check.